### ISSUE TITLE
Muti-Tenancy related query on openId auth connect

### ISSUE THREAD

**Issue by ashwinbasavaraja** [alerta/alerta#880]
Currently I am using openId connect for Auth added recently. 
But it has issue if the CUSTOMER_VIEWS =True as the customer lookup happens
be login = ashwin.basavaraja@gmail.com,[user] as highlighted in https://github.com/alerta/alerta/blob/master/alerta/auth/oidc.py#L53-L62

But the customer lookup happens if https://github.com/alerta/alerta/blob/master/alerta/database/backends/postgres/base.py#L1022-L1037
the match is happening by ashwin.basavaraja@gmail.com & next 'user'.Most of the time user is not allowed to login
if customer doesn't have either ashwin.basavaraja@gmail.com

So I do have 2 questions -
1. How to enable  CUSTOMER_VIEWS = True with OpenId auth connector?a    For Ex:- By domain abc.com
2. How to enable email separation at sub group level?
    For Ex:- finance team in abc.com, purchase team in abc.com so that respective team
    would see alert specific their departments

Any recommendation of supporting this is much appreciated

Thanks
Ashwin

**Comment by ashwinbasavaraja** [comment]
Also, Is it possible to look at the groups of Okta for 2nd usecase where customer is abc.com. The department like finance, purchase is group in Okta. This would help of alert look up at the sub group level

**Comment by satterly** [comment]
The "user" value is just the default role if no role is found in the `OIDC_CUSTOM_CLAIM` claim. What is your setting for `OIDC_CUSTOM_CLAIM`? What does your `userinfo` response look like? Can you provide an example? (make sure you remove any private data).

**Comment by ashwinbasavaraja** [comment]
From UI, we will use read & write scope. So the User management is completely outside the Alerta (as discussed in OpenId related thread). We don't do role mapping as it is defaulted to read & write. 
Sample User info payload looks like 

```
{'email': 'ashwin.basavaraja@abcd.com',
 'email_verified': True,
 'family_name': 'basavaraja',
 'given_name': 'ashwin',
 'groups': ['finance_user', 'purchace_user'],
 'locale': 'en-US',
 'name': 'ashwin basavaraja',
 'preferred_username': 'ashwin.basavaraja@abcd.com',
 'sub': '00....',
 'updated_at': ,
 'zoneinfo': 'America/Los_Angeles'}

```
Now we need to fetch the all the alerts belongs to 'finance_user', 'purchace_user' group even though the customer is abcd.com

Other scenario - If the groups attribute has only 'finance_user'. Need to pull alerts specific to finance_group.

**Comment by satterly** [comment]
Are you saying you would like to use the `groups` claim to preset a "group" filter that can be used by the web UI to only show certain alerts to certain users? ie. if a user logs in and they have `finance_user` in the userinfo `groups` payload then they should only see `finance_user` alerts (or better yet `finance_group` alerts)?

If that's what you're asking for then I can confirm that this isn't currently possible but it is on the roadmap as it's been asked for before. It should be noted that this would not prevent the user from changing the filter and seeing other groups alerts if they wanted to.

**Comment by ashwinbasavaraja** [comment]
Yes, I am looking for the feature.

Just asking on feasibility, of achieving this by making departments at the customer level. For Ex:- Instead of registering customer as abc.com. 

I will register 2 customers as finance_user & purchace_user. 

If the alerts are tagged properly to the group level while inserting - alerts related to finace_user will be tagged as customer as finance_user with key & similarly with purchase_user.

Alert view will display finance related alerts if the user group belongs to finance_user.

**Comment by satterly** [comment]
Ok. I think you're actually asking for the "customer" for a user to be configurable based on information in the userinfo. That is slightly different but I think possible. Let me look into it and get back to you.

**Comment by satterly** [comment]
Following #904 it is now possible to use a configurable JWT claim to assign customers to users. The server setting is `OIDC_GROUP_CLAIM` and defaults to `groups`. This will be available in version 7.0 which will be released in the next few weeks.
